screeps-game-api/src/objects/mod.rs

```diff
@@ -119,6 +119,44 @@ where
     }
 }
 
+/// Trait covering all objects with an id.
+pub unsafe trait HasId: AsRef<Reference> + RoomObjectProperties {
```

Comment:
I'd forgotten this before but I think `AsRef<Reference>` is now redundant because `RoomObjectProperties` also requires it.